sql/modules/Reconciliation.sql

```diff
@@ -93,6 +112,28 @@ $$Sets which lines of the report are cleared.$$;
 CREATE OR REPLACE FUNCTION reconciliation__delete_my_report(in_report_id int)
 RETURNS BOOL AS
 $$
+BEGIN
+    PERFORM approved FROM cr_report
+    WHERE id = in_report_id AND approved;
+    IF FOUND THEN
+        RAISE EXCEPTION 'Cannot delete approved recon report';
+        RETURN FALSE;
+    END IF;
+
+    -- Make sure that transactions present on this report do not have a
+    -- cleared_on date or they won't ever be accessible anymore.
```

Comment:
Of course the transactions are already cleared?! I mean, we're DELETING the report, right?

Reply:
Or should it be impossible to delete a report which has at least 1 cleared transaction?

Reply:
A transaction, not an entry line, gets cleared only when the reconciliation report is approved. If we ever get an already cleared transaction in a report that we are deleting, it won't ever be reconciliable again because we only select uncleared transactions for entry lines.

Reply:
I have no idea why you say the entire transaction will be cleared. If you look at https://github.com/ledgersmb/LedgerSMB/blob/master/sql/modules/Reconciliation.sql#L247-L248 you see that the `acc_trans` lines get set to cleared. Those are the acc_trans lines *included in the report*, which almost certainly means "not the entire transaction". 

Regardless, I don't understand how that's related to my question: if the report is approved, it has all cleared transaction lines. If the report is not approved, no transaction lines should be cleared. So the question then becomes: how is it possible that some lines have been marked as cleared while the report hasn't been approved?